### PR TITLE
catppuccin-gtk: fix nix derivation for build scripts

### PR DESCRIPTION
**Pull request by daylinmorgan**
## Description of changes

The previous derivation wouldn't actually run the build scripts or support multiple accent colors (since it expects the source to be reset with `git` after each theme generation).

- Patch build.sh shebang in colloid

- Add patch for `create_theme.py` to emulate git reset

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fufexan — APPROVED** [review]

File: pkgs/data/themes/catppuccin-gtk/create-theme.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/scripts/create_theme.py b/scripts/create_theme.py
+index 074dc43..7e7a8eb 100644
+--- a/scripts/create_theme.py
++++ b/scripts/create_theme.py
+@@ -10,7 +10,7 @@ from scripts.utils import replacetext, zip_multiple_folders
+ from scripts.var import def_color_map, repo_dir, src_dir, theme_name, work_dir
+ 
+ 
+-def create_theme(types: List[str], accents: List[str], dest: str, link: bool = False, 
++def create_theme(types: List[str], accents: List[str], dest: str, link: bool = False,
```
Comment (AndersonTorres) [inline pkgs/data/themes/catppuccin-gtk/create-theme.patch:10]: What does this line do?

File: pkgs/data/themes/catppuccin-gtk/create-theme.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     name: str = theme_name, size: str = "standard", tweaks=[], zip = False, recreate_assets = False) -> None:
+ 
+     try:
+@@ -34,7 +34,11 @@ def create_theme(types: List[str], accents: List[str], dest: str, link: bool = F
+             os.chdir(work_dir)
+             subprocess.call("./build.sh", shell=True) # Rebuild all scss
+             subprocess.call(install_cmd, shell=True) # Install the theme globally for you
+-            subprocess.call("git reset --hard HEAD", shell=True)  # reset colloid repo to original state
++            # subprocess.call("git reset --hard HEAD", shell=True)  # reset colloid repo to original state
```
Comment (AndersonTorres) [inline pkgs/data/themes/catppuccin-gtk/create-theme.patch:19]: This part is purely nix-specific, I suppose?

File: pkgs/data/themes/catppuccin-gtk/default.nix [hunk 3]
```diff
@@ -40,6 +40,10 @@ stdenvNoCC.mkDerivation rec {
 
   nativeBuildInputs = [ gtk3 sassc ];
 
+  patches = [
+    ./create-theme.patch
```
Comment (AndersonTorres) [inline pkgs/data/themes/catppuccin-gtk/default.nix:44]: Give a more descriptive name for this patch, not the mere filename being modified.

File: pkgs/data/themes/catppuccin-gtk/default.nix [hunk 4]
```diff
@@ -55,6 +59,7 @@ stdenvNoCC.mkDerivation rec {
 
   postPatch = ''
     patchShebangs --build colloid/install.sh
```
Comment (SuperSandro2000) [inline pkgs/data/themes/catppuccin-gtk/default.nix:61]: This can be combined into one

**Comment by SuperSandro2000** [comment]
> The previous derivation wouldn't actually run the build scripts or support multiple accent colors (since it expects the source to be reset with `git` after each theme generation).

The commit message should contain this in short form.

File: pkgs/data/themes/catppuccin-gtk/create-theme.patch (hunk 1 above)
Comment (daylinmorgan) [reply pkgs/data/themes/catppuccin-gtk/create-theme.patch]: Nothing, my editor had cleaned up whitespace seemingly, I removed this from the patch.

File: pkgs/data/themes/catppuccin-gtk/create-theme.patch (hunk 2 above)
Comment (daylinmorgan) [reply pkgs/data/themes/catppuccin-gtk/create-theme.patch]: That's correct the build scripts assumes colloid is a submodule but this derivation uses `colloid-gtk-theme` as an input.

**Comment by daylinmorgan** [comment]
I think I've addressed the comments from @SuperSandro2000 @AndersonTorres . 

Let me know if there is anything else that should be changed.

**Review by AndersonTorres — APPROVED** [review]
LGTM

**Pull request merged**